Source/ProjectFirma.Web/Views/Program/ProjectListGridSpec.cs

```diff
@@ -0,0 +1,65 @@
+﻿using ProjectFirma.Web;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using LtInfo.Common;
+using LtInfo.Common.DhtmlWrappers;
+using LtInfo.Common.HtmlHelperExtensions;
+using LtInfo.Common.ModalDialog;
+using LtInfo.Common.Views;
+using ProjectFirma.Web.Common;
+using ProjectFirma.Web.Controllers;
+using ProjectFirma.Web.Models;
+using ProjectFirma.Web.Security;
+
+namespace ProjectFirma.Web.Views.Program
+{
+    public class ProjectListGridSpec : GridSpec<Models.Project>
+    {
+        public ProjectListGridSpec(Person currentPerson, Models.Program currentProgram, Dictionary<int, List<Models.Program>> programsByProject)
+        {
+            var hasProgramManagePermissions = new ProgramManageFeature().HasPermissionByPerson(currentPerson);
+
+            if (hasProgramManagePermissions)
+            {
+                Add(string.Empty, x => x.ProjectImportBlockLists.Any(b => b.ProgramID == currentProgram.ProgramID)

```

Comment:
To make this more readable you could throw this into a lil private helper function like the "Program()" one that just returns the results of the modal dialog form link